src/mono/mono/mini/mini-llvm.c

```diff
@@ -7950,6 +7950,10 @@ MONO_RESTORE_WARNING
 			case XBINOP_FORCEINT_XOR:
 				result = LLVMBuildXor (builder, lhs_int, rhs_int, "");
 				break;
+			case XBINOP_FORCEINT_ANDN:
+				rhs_int = LLVMBuildNot (builder, rhs_int, "");
+				result = LLVMBuildAnd (builder,  lhs_int, rhs_int, "");
```

Comment:
Does LLVM optimize this down to `andnps/andnpd` for float/double?

Reply:
Actually, there is slight problem with this which I'm not sure what is causing it. For bitwise operations (BitwiseAnd, AndNot, etc.) implemented in Mono, LLVM is emitting for Vector128<T> only single precision floating point instruction, in this case `vandnps`.  I think numerically the results should be correct even for integer inputs, but I doubt it is the most effective solution for the CPU. When I examined the LLVM IR generated for integer inputs I didn't see any conversion to floats. I opened an issue for this https://github.com/dotnet/runtime/issues/74661.

Reply:
It matters less on modern CPUs and LLVM will generally do the "right" thing so provided Mono isn't itself saying they are all float, it's probably fine.

On some older or lower power devices, these could actually have different execution units and therefore using `andps` for integrals can cause a pipeline delay.